seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/config/PaimonConfig.java

```diff
@@ -61,9 +74,54 @@ public class PaimonConfig implements Serializable {
                     .noDefaultValue()
                     .withDescription("The read columns of the flink table store");
 
+    @Deprecated
     public static final Option<String> HDFS_SITE_PATH =
             Options.key("hdfs_site_path")
                     .stringType()
                     .noDefaultValue()
                     .withDescription("The file path of hdfs-site.xml");
+
+    public static final Option<Map<String, String>> HADOOP_CONF =
+            Options.key("paimon.hadoop.conf")
+                    .mapType()
+                    .defaultValue(new HashMap<>())
+                    .withDescription("Properties in hadoop conf");
+
+    public static final Option<String> HADOOP_CONF_PATH =
+            Options.key("paimon.hadoop.conf-path")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The specified loading path for the 'core-site.xml', 'hdfs-site.xml', 'hive-site.xml' files");
+
+    protected String catalogName;
+    protected String warehouse;
+    protected String namespace;
+    protected String table;
+    protected String hdfsSitePath;
+    protected Map<String, String> hadoopConfProps;
+    protected String hadoopConfPath;
+
+    public PaimonConfig(ReadonlyConfig readonlyConfig) {
+        this.catalogName = checkArgumentNotNull(readonlyConfig.get(CATALOG_NAME));
+        this.warehouse = checkArgumentNotNull(readonlyConfig.get(WAREHOUSE));
+        this.namespace = checkArgumentNotNull(readonlyConfig.get(DATABASE));
+        this.table = checkArgumentNotNull(readonlyConfig.get(TABLE));
+        this.hdfsSitePath = readonlyConfig.get(HDFS_SITE_PATH);
+        this.hadoopConfProps = readonlyConfig.get(HADOOP_CONF);
+        this.hadoopConfPath = readonlyConfig.get(HADOOP_CONF_PATH);
+    }
+
+    protected <T> T checkArgumentNotNull(T argument) {
+        checkNotNull(argument);
+        return argument;
+    }
+
+    @VisibleForTesting
```

Comment:
Can we remove this annotation? It is not intended for testing purposes.

Reply:
@Carl-Zhou-CN 
This annotation is meaningful to prevent method permissions from being changed.

Reply:
ok, it's my oversight